src/decode/libipt/mod.rs

```diff
@@ -0,0 +1,401 @@
+//! The libipt trace decoder.
+
+use crate::{c_errors::PerfPTCError, decode::TraceDecoder, errors::HWTracerError, Block, Trace};
+use libc::{c_char, c_int, c_void};
+use std::{convert::TryFrom, env, ffi::CString, os::fd::AsRawFd, ptr};
+use tempfile::NamedTempFile;
+
+extern "C" {
+    // decode.c
+    fn hwt_ipt_init_block_decoder(
+        buf: *const c_void,
+        len: u64,
+        vdso_fd: c_int,
+        vdso_filename: *const c_char,
+        decoder_status: *mut c_int,
+        err: *mut PerfPTCError,
+        current_exe: *const c_char,
+    ) -> *mut c_void;
+    fn hwt_ipt_next_block(
+        decoder: *mut c_void,
+        decoder_status: *mut c_int,
+        addr: *mut u64,
+        len: *mut u64,
+        err: *mut PerfPTCError,
+    ) -> bool;
+    fn hwt_ipt_free_block_decoder(decoder: *mut c_void);
+    // util.c
+    pub(crate) fn hwt_ipt_is_overflow_err(err: c_int) -> bool;
+    // libipt
+    pub(crate) fn pt_errstr(error_code: c_int) -> *const c_char;
+}
+
+pub(crate) struct LibIPTTraceDecoder {
+    //trace: &'t dyn Trace,
```

Comment:
Commented code.

Reply:
Fixed in [94cb536](https://github.com/ykjit/hwtracer/pull/99/commits/94cb536e85e8dbeb63fd23eae0d32389b5cbcf04)